processor/src/main/java/org/jboss/logging/processor/apt/TranslationClassGenerator.java

```diff
@@ -146,10 +147,18 @@ private List<File> findTranslationFiles(final MessageInterface messageInterface)
         if (translationFilesPath != null) {
             classTranslationFilesPath = translationFilesPath + packageName.replace('.', File.separatorChar);
 
-            //By default use the class output folder
+            //By default, use the class output folder
         } else {
-            FileObject fObj = processingEnv.getFiler().getResource(StandardLocation.CLASS_OUTPUT, packageName, interfaceName);
-            classTranslationFilesPath = fObj.toUri().getPath().replace(interfaceName, "");
+            // Create some random name:
+            String relativeName = interfaceName + UUID.randomUUID();
```

Comment:
What's weird to me is the exception is `java.io.FileNotFoundException`, yet using a random file name makes it work.

Reply:
Yeah, in the original approach, it was trying to get a resource (and eclipse compiler wants resources to actually exist) while this patch is just creating a new resource (so a potential problem could be that a file with that "random" name actually exists 🙈)